libs/wingc/src/type_check.rs

```diff
@@ -299,6 +302,27 @@ trait Subtype {
 	}
 }
 
+impl Subtype for Phase {
+	fn is_subtype_of(&self, other: &Self) -> bool {
+		// We model phase subtyping is as if the independent phase is an
+		// intersection type of preflight and inflight. This means that
+		// independent = preflight & inflight.
+		//
+		// This means the following pseudocode is valid:
+		// > let x: preflight fn = <phase-independent function>;
+		// (a phase-independent function is a subtype of a preflight function)
+		//
+		// But the following pseudocode is not valid:
+		// > let x: independent fn = <preflight function>;
+		// (a preflight function is not a subtype of an inflight function)
+		if self == &Phase::Independent {
+			true
+		} else {
+			self == other
+		}
+	}
+}
```

Comment:
I'm not convinced that `independent = preflight & inflight`, but I guess it's true enough currently so this is useful.

Reply:
Care to elaborate? (just curious - if it's a hunch, that's okay too :)) It might be relevant to the discussion on  #1711

Reply:
independent as an intersection type implies it has the functionality of both preflight and inflight, but it actually has neither. It's not "capturable" like preflight and it can't "capture" preflight things like inflight.

The fact that these aren't actually types makes the usage of subtyping feel off too. 
It kind feels similar to saying `non-reassignable` is a subtype of `reassignable`.

Reply:
I see. From a behavior perspective (what the code actually does) maybe it makes more sense to model this as union types (where given a type `T | U`, the only operations we can safely perform on it are ones that make sense for _both_ `T` and `U`)... Maybe I've just confused myself with the name "intersection type"?  😅